### PR TITLE
1315 - Fix add user agent detection for install button

### LINKED ISSUE #1257: Implement mobile install page
Blocks #1260
Blocks #1262
Blocks #1273

### LINKED ISSUE #1295: Install Page - Multi-line headlines need line-height increased
The headlines are feeling too tight here:
screen shot 2015-02-24 at 12 52 59 pm

### LINKED ISSUE #1296: Install Page - Extend the width / column length of subheads
The width of the subhead copy blocks is too narrow which is leading to some awkward line breaks:
screen shot 2015-02-24 at 12 52 59 pm

### LINKED ISSUE #1315: Port user agent logic to the install page

### PR DESCRIPTION
**Pull request by k88hudson**
fixes #1257
fixes #1315
fixes #1295
fixes #1296

### CODE REVIEW

File: views/app.html [hunk 1]
```diff
@@ -44,85 +44,77 @@
...
-        </div>
-        <div class="col-lg-12 pull-left">
-          <div class="col-lg-8 pull-left">
-            <div class="phone-error-popup off">{{ gettext("phone format error") }}</div>
-          </div>
-        </div>
-        <div class="col-lg-12 sms-privacy-message pull-left hidden">
-          <div class="col-lg-8 pull-left">
-            {{ gettext("sms privacy message") | safe }}
+{% macro preview(image) %}
```
Comment (secretrobotron) [inline views/app.html]: Make sure you include the `jpg` versions somehow. They're supposed to be in place of the gifs before they load.

File: views/app.html [hunk 2]
```diff
@@ -20,7 +20,7 @@
   <link rel="canonical" href="https://webmaker.org/{{ localeInfo.lang }}{{ currentPath }}" />
   <meta property="description" content="{{ gettext("Strips away complexity") }}" />
   <!-- /sharing -->
-
+  <link href='http://fonts.googleapis.com/css?family=Fira+Sans:400,700' rel='stylesheet' type='text/css'>
```
Comment (cadecairos) [inline views/app.html]: can we do this without using google fonts?

File: views/app.html [hunk 3]
```diff
@@ -44,85 +44,77 @@
...
+  <div class="flex-container">
+    <div class="left">
+      {{ preview('/img/app-install/templates@2x.gif') }}
+    </div>
+    <div class="right">
+      <h1>{{ gettext("Become a Maker") }}</h1>
+      <p class="lead">{{ gettext("Strips away complexity") }}</p>
+      <div class="nudge">
+        <button id="get-app" class="btn btn-neue desktop-only">
+          {{ gettext("Get App")}} <span class="fa fa-chevron-right"></span>
```
Comment (cadecairos) [inline views/app.html]: this button text isn't visible for me anymore

File: views/app.html [hunk 4]
```diff
@@ -44,85 +44,77 @@
...
+        </button>
+        <a href="http://mzl.la/installwm" class="btn btn-neue mobile-only">
+          {{ gettext("Download Webmaker")}}
+        </a>
+        <div id="sms-form" class="hidden">
+          <p class="text-left">{{ gettext("Enter your phone number")}}</p>
+          <div class="input-group">
+            <input class="form-control phone-num-input" type="tel" name="" value="">
+            <span class="input-group-btn">
+              <button class="btn btn-neue send-sms-btn">{{ gettext("Send Text")}}</button>
```
Comment (cadecairos) [inline views/app.html]: text not visible

File: views/app.html [hunk 5]
```diff
@@ -44,85 +44,77 @@
...
+      <p class="lead">{{ gettext("Strips away complexity") }}</p>
+      <div class="nudge">
+        <button id="get-app" class="btn btn-neue desktop-only">
+          {{ gettext("Get App")}} <span class="fa fa-chevron-right"></span>
+        </button>
+        <a href="http://mzl.la/installwm" class="btn btn-neue mobile-only">
+          {{ gettext("Download Webmaker")}}
+        </a>
+        <div id="sms-form" class="hidden">
+          <p class="text-left">{{ gettext("Enter your phone number")}}</p>
```
Comment (cadecairos) [inline views/app.html]: missing the phone icon from font-awesome

File: views/app.html [hunk 6]
```diff
@@ -44,85 +44,77 @@
...
-          <div class="col-lg-4 pull-left">
-            <input class="phone-num-input" type="tel" name="" value="">
-          </div>
-          <div class="col-lg-4">
-            <button class="btn btn-block send-sms-btn">{{ gettext("Send Text")}}</button>
-          </div>
-          <div class="col-lg-12 message-sent-container hidden pull-left">
-            <span>{{ gettext("message sent") }}</span>
-          </div>
-          <div class="col-lg-12 message-sent-error-container hidden pull-left">
```
Comment (cadecairos) [inline views/app.html]: this wasn't added back in

File: views/app.html [hunk 7]
```diff
@@ -44,85 +44,77 @@
...
-            <span>{{ gettext("message sent error") }}</span>
-          </div>
-        </div>
-        <div class="col-lg-12 pull-left">
-          <div class="col-lg-8 pull-left">
-            <div class="phone-error-popup off">{{ gettext("phone format error") }}</div>
```
Comment (cadecairos) [inline views/app.html]: this was also not added back in

File: views/app.html (hunk 3 above)
Comment (secretrobotron) [inline views/app.html]: Yeah, watch your spacing.
`{{ gettext("Get App")}}` vs
`{{ gettext("Get App") }}`

File: views/app.html (hunk 4 above)
Comment (secretrobotron) [inline views/app.html]: spacing

File: views/app.html (hunk 5 above)
Comment (secretrobotron) [inline views/app.html]: spacing

Comment (k88hudson) [inline views/app.html]: personally I think we should remove the icon, need to discuss

File: views/app.html (hunk 6 above)
Comment (k88hudson) [inline views/app.html]: oops! good catch :P

File: views/app.html (hunk 2 above)
Comment (k88hudson) [inline views/app.html]: hm, no real easy way to do this right now without adding a huge amount of page weight to the whole repo. can we file as a follow up?

File: views/app.html (hunk 1 above)
Comment (k88hudson) [inline views/app.html]: can we file as a follow up?

File: views/app.html (hunk 5 above)
Comment (k88hudson) [inline views/app.html]: ok, looks like we're good to remove it

File: views/app.html (hunk 2 above)
Comment (cadecairos) [inline views/app.html]: Fira fonts are already in https://github.com/mozilla/webmaker.org/tree/master/public/resources/fonts and imported in less

**Comment by k88hudson** [comment]
@cadecairos @secretrobotron ok review changes are in, except for embedding fira and jpg fallabcks for preloading. can we file those as separate issues so we can get this on staging as soon as possible for QA?

**Comment by cadecairos** [comment]
The sub-heading text is missing!

File: views/app.html (hunk 5 above)
Comment (secretrobotron) [inline views/app.html]: spacing?

```
{{ gettext("Enter your phone number")>>>>>>>RIGHT HERE<<<<<<<}}
```

File: views/app.html (hunk 1 above)
Comment (secretrobotron) [inline views/app.html]: Yeah, no problem. Go ahead.

**Comment by secretrobotron** [comment]
@k88hudson this is fine. There are a few missing things now though, like the error message and red textbox coloring when your phone number isn't correct. Need to fix them here if it's easy to drop that code back in. Otherwise, file a few followups and let's keep this ship sailing.

**Comment by cadecairos** [comment]
fonts look good now. Still missing the error message for when you enter an invalid phone number (unless you're going to add it back in in a followup)

**Comment by k88hudson** [comment]
Ok final changes in!

**Comment by cadecairos** [comment]
R+

**Pull request merged**